spec/models/query_log_spec.rb

```diff
@@ -0,0 +1,48 @@
+require 'query_log'
+
+RSpec.describe QueryLog, type: :model do
+  describe '.start' do
+    it 'initializes an instance with params and start time' do
+      ctdl = { "Query" => { "@type" => "ceterms:Certificate", "ceterms:name" => "accounting" } }
+      query_log = QueryLog.start(
+        engine: 'sparql',
+        ctdl: ctdl
+      )
+      expect(query_log.started_at).not_to be_nil
+      expect(query_log.completed_at).to be_nil
+      expect(query_log.engine).to eq('sparql')
+      expect(query_log.ctdl).to eq(ctdl)
+      expect(query_log.persisted?).to be true
+    end
+  end
+
+  describe '#complete' do
+    it 'updates the completed time' do
+      query_log = QueryLog.start(
+        engine: 'sparql',
+        ctdl: { "Query" => { "@type" => "ceterms:Certificate", "ceterms:name" => "accounting" } }
+      )
+      expect(query_log.completed_at).to be_nil
+      query_log.complete
```

Comment:
The `QueryLog#complete` method expects one argument. Maybe it should have a default value?

Reply:
Oops, looks like I forgot to update the spec. I think it's better if we make it required here - if the operation completed successfully, there must be a result for `complete`, right? `fail` should be used otherwise.